syntax = "proto3";
package turbomessage;

service Authenticator{
	rpc new_user(New_User_Data) returns (Status){};
	rpc authenticate(User_Data) returns (Status){};
}

service EmailServer{
	rpc send_email(Email_Data) returns (Status){};
	rpc recive_emails(User_Email) returns(stream Email_Response){};
	rpc mark_as_read(Email_Data) returns(Status){};
}

message New_User_Data{
	optional string name = 1;
	optional string last_name = 2;
	optional string email = 3;
	optional string password = 4;
}

message User_Data{
	optional string email = 1;
	optional string password = 2;
}

message Status{
	optional bool status = 1;
	optional string message = 2;
}

message Email_Data{
	optional int32 id  = 1;
	optional string subject = 2;
	optional bool read = 3;
	optional bool deleted = 4;
	optional string email_sender = 5;
	optional string email_reciver = 6;
}

message User_Email{
	string email = 1;
}

message Email_Response{
	repeated Email_Data emails = 1;
}